lightning/src/sign/ecdsa.rs

```diff
@@ -209,4 +209,15 @@ pub trait EcdsaChannelSigner: ChannelSigner {
 	fn sign_channel_announcement_with_funding_key(
 		&self, msg: &UnsignedChannelAnnouncement, secp_ctx: &Secp256k1<secp256k1::All>,
 	) -> Result<Signature, ()>;
+	/// Sign an input of a transaction with our funding key.
+	/// Used for splicing, when signing the previous funding transaction.
+	/// The previous funding transaction becomes an input to the new funding transaction,
+	/// and it is a multisig, which we also need to sign.
```

Comment:
The wording here is a bit confusing because the first sentence says "sign and input" but the second sentence you're saying "signing the previous funding transaction". I think we can make this less verbose as:

> Signs the input of a splicing funding transaction with our funding key.
> 
> In splicing, the previous funding transaction output is spent as the input of the new funding transaction, and is a 2-of-2 multisig.